Lib/locale.py

```diff
@@ -1467,10 +1467,15 @@ def getpreferredencoding(do_setlocale=True):
 #
 
 windows_locale = {
```

Comment:
Would it be possible to call GetLocaleInfo() instead of using a hardcoded dictionary?

Reply:
It is called. The dictionary is used if `GetLocaleInfo` fails to provide such information (on Windows 95, etc). That is, almost never.

And if we use `GetLocaleInfo`, `windows_locale` is not needed.

There is also better modern API than `GetLocaleInfo`, but since `_getdefaultlocale()` is deprecated anyway...

Reply:
> It is called. The dictionary is used if GetLocaleInfo fails to provide such information (on Windows 95, etc)

We don't support Windows 95 anymore. Do you mean that the dict is no longer needed?

> And if we use GetLocaleInfo, windows_locale is not needed.

I don't understand. getdefaultlocale() calls GetLocaleInfo() on Windows.

Reply:
Yes, I think that it is no longer needed. `GetLocaleInfoA` supports LOCALE_SISO639LANGNAME and LOCALE_SISO3166CTRYNAME on all modern Windows. It only returns a LCID if:
```c
    /* If we end up here, this windows version didn't know about
       ISO639/ISO3166 names (it's probably Windows 95).  Return the
       Windows language identifier instead (a hexadecimal number) */
```
And only in this case the dict is used.